### DISCUSSION TITLE
Seeking Advice on migrating my project from pimcore 6.6 to pimcore 11.3

### DISCUSSION THREAD

**Discussion by anshu1thakur** [pimcore/pimcore#17472] [Q&A / Support]
Hi everyone,

I am planning to migrate my project from pimcore 6.6 to pimcore 11.3(latest verision), and I would greatly appreciate any advice , suggestions,insights or experiences that would make this migration process as smooth as possible . 

Suggestions like how should I set up my bundles , any database change which I might miss or anything which might impact my process.
I will also share my experience in between or after I successfully complete it .

I would be greateful for any advice or experience you can share . Thank you in advance for help.

**Comment by dpfaffenbauer** [comment]
1. Update to 6.9.6 and run the migrations. Should be fairly easy
2. Update to 10.6 (follow Pimcore instructions) 
3. Test everything, lots changed in this version and therefore this is the hardest update
    - eg. Thumbnail Creation Twig function changed parameters for classnames
5. Update to 11.3 (follow Pimcore instructions)
6. Test everything, some things have changed, this update is easier then 6 to 10

Really, test everything, you will miss things. For that use some Error Monitoring Tool like Sentry to see whats going on. 

You also have to run these migrations and updates in Production step-by-step. There is no way of going from 6 all the way to 11 in one deployment.

**Comment by anshu1thakur** [comment]
Is there  any way to directly migrate from pimcore 6 to 11. As I can see that a lot of variables in the class definition ,that are changed and even there datatype are changed as well , So do I have to manually create ecah and every class from scratch or Is there any way to migrate it

**Comment by EliShklovsky** [comment]
Hi
In order to migrate classes from 6 to 11  - i.e export and import you have to open the exported json file in file editor and make some search-replace:

It is preferable to use the Bulk Export and Bulk Import.

childs
children

AppBundle\\
App\\

\\AppBundle\\
\\App\\

"maxItems": "",
"maxItems": null,

"rows": "",
"rows": null,

 "cols": "",
 "cols": null,

Erase completely:
 "fieldtype": "button",

Hope that helps.
